plugin/PySrc/code_tracer.py

```diff
@@ -49,6 +49,11 @@ class MetaPathFinder(object):
     from report_builder import ReportBuilder
     document = None
 
+try:
+    from mock_pyglet import MockPyglet
```

Comment:
This is fine for a proof of concept, but it forces an import of pyglet, even if the source code doesn't import it. See how the [TracedModuleImporter] handles matplotlib, and duplicate that for pyglet.

[TracedModuleImporter]: https://github.com/donkirkby/live-py-plugin/blob/v2.25.1/plugin/PySrc/code_tracer.py#L732

Reply:
Just having a look at this now. That would mean the call to monkey patch would probably sit under def exec_module(self, module). Does that sound right? I need an instance of the canvas to pass to monkey_patch - I could pass MockTurtle.cv potentially?

Reply:
That sounds like the right place to do the patching. You can see how to get at a canvas [here].

[here]: https://github.com/donkirkby/live-py-plugin/blob/v2.25.1/plugin/PySrc/code_tracer.py#L848-L849